src/main/java/edu/harvard/iq/dataverse/api/Access.java

```diff
@@ -1787,37 +1761,41 @@ private boolean isAccessAuthorized(DataFile df, String apiToken) {
             }
         }
         
-        if (!restricted && !embargoed) {
-            // And if they are not published, they can still be downloaded, if the user
+        
+
+        //The one case where we don't need to check permissions
+        if (!restricted && !embargoed && published) {
+            // If they are not published, they can still be downloaded, if the user
             // has the permission to view unpublished versions! (this case will 
             // be handled below)
-            if (published) { 
-                return true;
-            }
+            return true;
         }
         
+        //For permissions check decide if we havce a session user, or an API user
         User user = null;
         
         /** 
          * Authentication/authorization:
-         * 
-         * note that the fragment below - that retrieves the session object
-         * and tries to find the user associated with the session - is really
-         * for logging/debugging purposes only; for practical purposes, it 
-         * would be enough to just call "permissionService.on(df).has(Permission.DownloadFile)"
-         * and the method does just that, tries to authorize for the user in 
-         * the current session (or guest user, if no session user is available):
          */
         
-        if (session != null) {
+        User apiTokenUser = null;
+        //If we get a non-GuestUser from findUserOrDie, use it. Otherwise, check the session
+        try {
+            logger.fine("calling apiTokenUser = findUserOrDie()...");
+            apiTokenUser = findUserOrDie();
```

Comment:
With the rearchitecting, I assume this call will end up handling both session and token logins and this part of the code will be able to be replaced. Until then, this is the only part of the api that was already designed to be usable with sessions or tokens.